### PR TITLE
Refactors aruba cumcumber tests parallel

### PR DESCRIPTION
**Pull request by luispresuelVenafi**
Refactors aruba cucumber tests flow in order to allow different flow executions by each platform. This is in order to **allow parallel execution** for external call by each PLATFORM.

This includes:

- Modifying `cucumber.sh` script in order to take two named variables: "FEATURE" and "PLATFORM. Fixes behavior in order to properly verify if "GNU parallel" is installed.
- Modifying `Makefile` as well for this execution
- Modifying `Dockerfile` in order to allow this tags

**IMPORTANT**: Starting from this changes, you will need to tag every test according to each platform

`@TPP`: For Venafi Trust Protection Platform

`@VAAS`: Venafi as a Service

`@FAKE`: test mode (Fake connector)

`@COMMON`: Cases that impact VCert itself and is not tied to a platform (e.g. generating a CSR, checking VCert version, verifying VCert help, etc.)

#### For every test

`make cucumber`

#### For executing tests by feature

**FEATURE** should be the path to the feature, example:

`make cucumber FEATURE=features/retire/retire.feature`

#### For executing tests by platform

**PLATFORM** should be any of `TPP`, `VAAS`, `FAKE` or `COMMON`

**FAKE:** Handles test cases that are in test mode (Fake connector)

**COMMON**: Cases that impact VCert itself and is not tied to a platform (e.g. generating a CSR, checking VCert version, verifying VCert help, etc.)

example:

`make cucumber PLATFORM=VAAS` # will execute all VAAS related test`

#### For executing tests by feature and platform

`make cucumber FEATURE=features/retire/retire.feature PLATFORM=VAAS` # will execute only retire feature tests on VaaS

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,11 +35,18 @@ cucumber:
 	rm -rf ./aruba/bin/
 	mkdir -p ./aruba/bin/ && cp ./bin/linux/vcert ./aruba/bin/vcert
 	docker build --tag vcert.auto aruba/
-	if [ -z "$(FEATURE)" ]; then \
-		cd aruba && ./cucumber.sh; \
+	if [ -n "$(FEATURE)" ] && [ -n "$(PLATFORM)" ]; then \
+		echo "executing both feature and platform"; \
```
Comment (rvelaVenafi) [inline Makefile:39]: What feature, what platform?

File: aruba/features/config/step_definitions/my_steps.rb [hunk 2]
```diff
@@ -18,4 +18,11 @@
...
           """
         }
+end
+
+When(/^I enroll random certificate \-config "([^"]*)" \-profile (.*) with (.+)?$/) do |config, profile, flags|
+  cn = " -cn " + random_cn
+  cmd = "vcert enroll -config #{config} -profile #{profile} #{cn} #{flags}"
+
+  steps %{Then I try to run `#{cmd}`}
 end
```
Comment (rvelaVenafi) [inline aruba/features/config/step_definitions/my_steps.rb:28]: Missing EOF here

File: aruba/features/enroll/basic.enroll.feature [hunk 3]
```diff
@@ -102,10 +121,12 @@ Feature: Enroll certificate
...
 
+  @TPP
   Scenario: enroll with duplicated instance
     Given I enroll random certificate using TPP with -no-prompt -field "custom=12121" -field "Server Names=some server" -instance devops-instance:nginx_246 -instance devops-instance
     Then the exit status should be 1
 
+  @TPP
   Scenario: enroll with duplicated tls-address
     Given I enroll random certificate using TPP with -no-prompt -field "custom=12121" -field "Server Names=some server" -tls-address api-gw-myapp.example:8443 -tls-address api-gw-myapp.example
     Then the exit status should be 1
```
Comment (rvelaVenafi) [inline aruba/features/enroll/basic.enroll.feature:132]: Missing EOF here

File: aruba/features/renew/renew-with-csr-service.feature [hunk 4]
```diff
@@ -59,7 +60,10 @@ Feature: renew action with `-csr service` option
         And it should write certificate to the file "c1.pem"
         And private key in "k1.pem" and certificate in "c1.pem" should have the same modulus
 
-  @TODO # This scenario is not currently working due to reuse csr not being implemented yet in OutagePredict
+  @TODO @VAAS # This scenario is not currently working due to reuse csr not being implemented yet in OutagePredict
```
Comment (rvelaVenafi) [inline aruba/features/renew/renew-with-csr-service.feature:63]: If there is an update here, remove the previos comment as it no longer applies. It only makes explanation more complex than required.

**Review by rvelaVenafi — COMMENTED** [review]
Left comments

File: aruba/features/config/step_definitions/my_steps.rb (hunk 2 above)
Comment (luispresuelVenafi) [reply aruba/features/config/step_definitions/my_steps.rb:28]: This is not introduced by my changes. But sure, will fix it by next commit

File: aruba/features/renew/renew-with-csr-service.feature (hunk 4 above)
Comment (luispresuelVenafi) [reply aruba/features/renew/renew-with-csr-service.feature:63]: I mostly may need to edit it, no longer applies, but requires verification (hence not removed the "TODO"). Will clarify in next commit

**Review by rvelaVenafi — APPROVED** [review]

File: aruba/features/renew/renew-with-csr-service.feature (hunk 4 above)
Comment (luispresuelVenafi) [reply aruba/features/renew/renew-with-csr-service.feature]: Done by latest changes

File: aruba/features/enroll/basic.enroll.feature (hunk 3 above)
Comment (luispresuelVenafi) [reply aruba/features/enroll/basic.enroll.feature]: Done by latest changes

File: aruba/features/config/step_definitions/my_steps.rb (hunk 2 above)
Comment (luispresuelVenafi) [reply aruba/features/config/step_definitions/my_steps.rb]: Done by latest changes

File: Makefile (hunk 1 above)
Comment (luispresuelVenafi) [reply Makefile]: Added variable inside echo by latest changes

**Review by EduardoVV — APPROVED** [review]

**Pull request merged**